src/presentation/http/router/note.test.ts

```diff
@@ -239,12 +233,115 @@ describe('Note API', () => {
           url: `/note/${publicId}`,
         });
 
-        expect(response?.statusCode).toBe(expectedStatus);
+        expect(response?.statusCode).toBe(200);
 
         expect(response?.json().accessRights).toStrictEqual({ canEdit: true });
       });
     });
+  });
+
+  describe('PATCH note/:notePublicId ', () => {
+    test('Update note by public id with 200 status, user is creator of the note', async () => {
+      const userId = 2;
+      const accessToken = global.auth(userId);
+
+      const userNote = notes.find(newNote => {
+        return newNote.creator_id === userId;
+      });
+
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${userNote!.public_id}`,
+        body: {
+          'content': { new: 'content added' },
+        },
+      });
+
+      expect(response?.statusCode).toBe(200);
+    });
+
+    test('Returns status 401 when the user is not authorized', async () => {
+      const correctID = 'Pq1T9vc23Q';
+
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        url: `/note/${correctID}`,
+        body: {},
+      });
+
+      expect(response?.statusCode).toBe(401);
+
+      expect(response?.json()).toStrictEqual({ message: 'You must be authenticated to access this resource' });
+    });
+
+    test('Returns status 406 when the public id does not exist', async () => {
+      const nonexistentId = 'ishvm5qH84';
+
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        url: `/note/${nonexistentId}`,
+        body: {},
+      });
+
+      expect(response?.statusCode).toBe(406);
+
+      expect(response?.json()).toStrictEqual({ message: 'Note not found' });
+    });
+
+    test.each([
+      { id: 'mVz3iHuez',
+        expectedMessage: 'params/notePublicId must NOT have fewer than 10 characters' },
+
+      { id: 'cR8eqF1mFf0',
+        expectedMessage: 'params/notePublicId must NOT have more than 10 characters' },
+
+      { id: '+=*&*5%&&^&-',
+        expectedMessage: '\'/note/+=*&*5%&&^&-\' is not a valid url component' },
+    ])
+    ('Returns 400 when public id of the note settings has incorrect characters and length', async ({ id, expectedMessage }) => {
+      const response = await global.api?.fakeRequest({
+        method: 'PATCH',
+        url: `/note/${id}`,
+      });
+
+      expect(response?.statusCode).toBe(400);
+
+      expect(response?.json().message).toStrictEqual(expectedMessage);
+    });
+
+    test.todo('Returns 400 when parentId has incorrect characters and lenght');
+  });
+
+  describe('POST /note', () => {
+    test('Returns 200 status, existing parentId passed in the body', async () => {
```

Comment:
Why do you check for a code, not for an actual behaviour? You need to check that added note has passed note as a child